Runtime error when "vault auth" against revoked token

I ran `vault token-create` and received a token. I then ran `vault auth TOKEN_ID` and received a "Successfully authenticated!" message. Finally I ran `vault token-revoke TOKEN_ID` and received a "Revocation successful" message. So far so good.

I then ran another `vault auth TOKEN_ID` again to see what error I would receive for an invalid auth token, and received the following error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x4a5765]

goroutine 1 [running]:
github.com/hashicorp/vault/command.(*AuthCommand).Run(0xc20808e200, 0x0, 0x0, 0x0, 0xc208042380)
        /Users/mitchellh/code/go/src/github.com/hashicorp/vault/command/auth.go:156 +0x1665
github.com/mitchellh/cli.(*CLI).Run(0xc208044b40, 0xc20801e640, 0x0, 0x0)
        /Users/mitchellh/code/go/src/github.com/mitchellh/cli/cli.go:100 +0x3ce
github.com/hashicorp/vault/cli.RunCustom(0xc20800a010, 0x2, 0x2, 0xc20800b1a0, 0xa1a080)
        /Users/mitchellh/code/go/src/github.com/hashicorp/vault/cli/main.go:44 +0x484
github.com/hashicorp/vault/cli.Run(0xc20800a010, 0x2, 0x2, 0x5055d0)
        /Users/mitchellh/code/go/src/github.com/hashicorp/vault/cli/main.go:11 +0x53
main.main()
        /Users/mitchellh/code/go/src/github.com/hashicorp/vault/main.go:10 +0x6d

goroutine 5 [syscall]:
os/signal.loop()
        /Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 6 [semacquire]:
sync.(*Cond).Wait(0xc2080c8030)
        /Users/mitchellh/code/3rdparty/go/src/sync/cond.go:62 +0x9e
io.(*pipe).read(0xc2080c8000, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /Users/mitchellh/code/3rdparty/go/src/io/pipe.go:52 +0x303
io.(*PipeReader).Read(0xc208036058, 0xc20800f000, 0x1000, 0x1000, 0xb9f630, 0x0, 0x0)
        /Users/mitchellh/code/3rdparty/go/src/io/pipe.go:134 +0x5b
bufio.(*Scanner).Scan(0xc20808e280, 0x48c17b)
        /Users/mitchellh/code/3rdparty/go/src/bufio/scan.go:180 +0x688
github.com/hashicorp/vault/command.func·010()
        /Users/mitchellh/code/go/src/github.com/hashicorp/vault/command/meta.go:176 +0x3d
created by github.com/hashicorp/vault/command.(*Meta).FlagSet
        /Users/mitchellh/code/go/src/github.com/hashicorp/vault/command/meta.go:179 +0x4c1

goroutine 15 [runnable]:
net/http.(*persistConn).readLoop(0xc2080c60b0)
        /Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
        /Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:660 +0xc9f

goroutine 16 [select]:
net/http.(*persistConn).writeLoop(0xc2080c60b0)
        /Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
        /Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:661 +0xcbc
```

--- a/command/auth.go
+++ b/command/auth.go
@@ -151,6 +151,10 @@ func (c *AuthCommand) Run(args []string) int {
 			"Error validating token: %s", err))
 		return 1
 	}
+	if secret == nil {
+		c.Ui.Error(fmt.Sprintf("Error: Invalid token"))
+		return 1
+	}
 
 	// Get the policies we have
 	policiesRaw, ok := secret.Data["policies"]

--- a/command/auth_test.go
+++ b/command/auth_test.go
@@ -82,6 +82,29 @@ func TestAuth_token(t *testing.T) {
 	}
 }
 
+func TestAuth_badToken(t *testing.T) {
+	core, _, _ := vault.TestCoreUnsealed(t)
+	ln, addr := http.TestServer(t, core)
+	defer ln.Close()
+
+	testAuthInit(t)
+
+	ui := new(cli.MockUi)
+	c := &AuthCommand{
+		Meta: Meta{
+			Ui: ui,
+		},
+	}
+
+	args := []string{
+		"-address", addr,
+		"not-a-valid-token",
+	}
+	if code := c.Run(args); code != 1 {
+		t.Fatalf("bad: %d\n\n%s", code, ui.ErrorWriter.String())
+	}
+}
+
 func TestAuth_method(t *testing.T) {
 	core, _, token := vault.TestCoreUnsealed(t)
 	ln, addr := http.TestServer(t, core)

